Hola **Luis**!

Soy **Patricio Requena** 👋. Es un placer ser el revisor de tu proyecto el día de hoy!

Revisaré tu proyecto detenidamente con el objetivo de ayudarte a mejorar y perfeccionar tus habilidades. Durante mi revisión, identificaré áreas donde puedas hacer mejoras en tu código, señalando específicamente qué y cómo podrías ajustar para optimizar el rendimiento y la claridad de tu proyecto. Además, es importante para mí destacar los aspectos que has manejado excepcionalmente bien. Reconocer tus fortalezas te ayudará a entender qué técnicas y métodos están funcionando a tu favor y cómo puedes aplicarlos en futuras tareas. 

_**Recuerda que al final de este notebook encontrarás un comentario general de mi parte**_, empecemos!

Encontrarás mis comentarios dentro de cajas verdes, amarillas o rojas, ⚠️ **por favor, no muevas, modifiques o borres mis comentarios** ⚠️:


<div class="alert alert-block alert-success">
<b>Comentario del revisor</b> <a class=“tocSkip”></a>
Si todo está perfecto.
</div>

<div class="alert alert-block alert-warning">
<b>Comentario del revisor</b> <a class=“tocSkip”></a>
Si tu código está bien pero se puede mejorar o hay algún detalle que le hace falta.
</div>

<div class="alert alert-block alert-danger">
<b>Comentario del revisor</b> <a class=“tocSkip”></a>
Si de pronto hace falta algo o existe algún problema con tu código o conclusiones.
</div>

Puedes responderme de esta forma:
<div class="alert alert-block alert-info">
<b>Respuesta del estudiante</b> <a class=“tocSkip”></a>


# Librerias

In [9]:
import pandas as pd
import numpy as np
from sklearn.impute import SimpleImputer
from sklearn.model_selection import train_test_split
from sklearn.compose import ColumnTransformer
from sklearn.preprocessing import OneHotEncoder, StandardScaler
from sklearn.pipeline import Pipeline
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import f1_score, roc_auc_score
from sklearn.ensemble import RandomForestClassifier


In [10]:
from sklearn.utils import resample

In [11]:

from sklearn.utils import shuffle

In [18]:
from sklearn.metrics import f1_score, roc_auc_score, classification_report

# 1.Obserbación y limpieza de datos:

In [2]:
df = pd.read_csv('/datasets/Churn.csv')

In [3]:
print(df.info())
print(df.head())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   RowNumber        10000 non-null  int64  
 1   CustomerId       10000 non-null  int64  
 2   Surname          10000 non-null  object 
 3   CreditScore      10000 non-null  int64  
 4   Geography        10000 non-null  object 
 5   Gender           10000 non-null  object 
 6   Age              10000 non-null  int64  
 7   Tenure           9091 non-null   float64
 8   Balance          10000 non-null  float64
 9   NumOfProducts    10000 non-null  int64  
 10  HasCrCard        10000 non-null  int64  
 11  IsActiveMember   10000 non-null  int64  
 12  EstimatedSalary  10000 non-null  float64
 13  Exited           10000 non-null  int64  
dtypes: float64(3), int64(8), object(3)
memory usage: 1.1+ MB
None
   RowNumber  CustomerId   Surname  CreditScore Geography  Gender  Age  \
0          1   

In [4]:


# 1) Cargar datos
df = pd.read_csv('/datasets/Churn.csv')

# 2) Dimensión y columnas
print("Shape:", df.shape)
print("\nColumnas:")
print(df.columns.tolist())

# 3) Vista rápida
print("\nHead:")
display(df.head(3))
print("\nTail:")
display(df.tail(3))

# 4) Tipos + nulos
print("\nInfo:")
print(df.info())
print("\nNulos por columna:")
print(df.isnull().sum().sort_values(ascending=False))

# 5) Duplicados
print("\nFilas duplicadas:", df.duplicated().sum())



Shape: (10000, 14)

Columnas:
['RowNumber', 'CustomerId', 'Surname', 'CreditScore', 'Geography', 'Gender', 'Age', 'Tenure', 'Balance', 'NumOfProducts', 'HasCrCard', 'IsActiveMember', 'EstimatedSalary', 'Exited']

Head:


RowNumber  CustomerId   Surname  CreditScore Geography  Gender  Age  \
0          1    15634602  Hargrave          619    France  Female   42   
1          2    15647311      Hill          608     Spain  Female   41   
2          3    15619304      Onio          502    France  Female   42   

   Tenure    Balance  NumOfProducts  HasCrCard  IsActiveMember  \
0     2.0       0.00              1          1               1   
1     1.0   83807.86              1          0               1   
2     8.0  159660.80              3          1               0   

   EstimatedSalary  Exited  
0        101348.88       1  
1        112542.58       0  
2        113931.57       1


Tail:


RowNumber  CustomerId    Surname  CreditScore Geography  Gender  Age  \
9997       9998    15584532        Liu          709    France  Female   36   
9998       9999    15682355  Sabbatini          772   Germany    Male   42   
9999      10000    15628319     Walker          792    France  Female   28   

      Tenure    Balance  NumOfProducts  HasCrCard  IsActiveMember  \
9997     7.0       0.00              1          0               1   
9998     3.0   75075.31              2          1               0   
9999     NaN  130142.79              1          1               0   

      EstimatedSalary  Exited  
9997         42085.58       1  
9998         92888.52       1  
9999         38190.78       0


Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   RowNumber        10000 non-null  int64  
 1   CustomerId       10000 non-null  int64  
 2   Surname          10000 non-null  object 
 3   CreditScore      10000 non-null  int64  
 4   Geography        10000 non-null  object 
 5   Gender           10000 non-null  object 
 6   Age              10000 non-null  int64  
 7   Tenure           9091 non-null   float64
 8   Balance          10000 non-null  float64
 9   NumOfProducts    10000 non-null  int64  
 10  HasCrCard        10000 non-null  int64  
 11  IsActiveMember   10000 non-null  int64  
 12  EstimatedSalary  10000 non-null  float64
 13  Exited           10000 non-null  int64  
dtypes: float64(3), int64(8), object(3)
memory usage: 1.1+ MB
None

Nulos por columna:
Tenure             909
RowNumber            0
CustomerId      

In [5]:



df = pd.read_csv('/datasets/Churn.csv')

# 1) descartar columnas de identificación
df = df.drop(columns=['RowNumber', 'CustomerId', 'Surname'])

# 2) imputar Tenure (años) y convertir a entero
imp = SimpleImputer(strategy='most_frequent')  # o 'median'
df['Tenure'] = imp.fit_transform(df[['Tenure']]).ravel().astype(int)

# 3) opcional: castear binarios a bool
for col in ['HasCrCard', 'IsActiveMember', 'Exited']:
    df[col] = df[col].astype(bool)

# 4) opcional: marcar categóricas como category (para EDA/memoria)
df['Geography'] = df['Geography'].astype('category')
df['Gender'] = df['Gender'].astype('category')

print(df.dtypes)
print(df.isnull().sum())


CreditScore           int64
Geography          category
Gender             category
Age                   int64
Tenure                int64
Balance             float64
NumOfProducts         int64
HasCrCard              bool
IsActiveMember         bool
EstimatedSalary     float64
Exited                 bool
dtype: object
CreditScore        0
Geography          0
Gender             0
Age                0
Tenure             0
Balance            0
NumOfProducts      0
HasCrCard          0
IsActiveMember     0
EstimatedSalary    0
Exited             0
dtype: int64


Se ajustaron los tipos de datos para mejorar coherencia, eficiencia y preparación del modelado. Se eliminaron columnas de identificación (RowNumber, CustomerId, Surname) por no aportar señal predictiva y para evitar fuga de información. Las variables categóricas Geography y Gender se tiparon como category para reflejar su naturaleza discreta y facilitar su codificación OHE sin errores. Los indicadores HasCrCard, IsActiveMember y el objetivo Exited se convirtieron a bool, lo que clarifica su semántica binaria y reduce ambigüedad frente a enteros 0/1. Tenure (años) se imputó y casteó a int, pues representa conteos enteros; el resto de numéricas permanecieron como int64/float64 según corresponda (CreditScore, Age, NumOfProducts, Balance, EstimatedSalary). Tras estos cambios, no quedaron valores nulos, y el dataset quedó listo para un pipeline de modelado (OHE para categóricas y, si se requiere, escalado para modelos lineales) con tipos más compactos y consistentes.

# Punto 2

## 2.1 Equilibrio de Clases

In [6]:
# 1) Balance del target
print("\nBalance de clases (Exited):")
print(df['Exited'].value_counts())
print("\nBalance relativo (Exited):")
print(df['Exited'].value_counts(normalize=True).round(3))

# 2) Estadísticas numéricas
num_cols = df.select_dtypes(include=[np.number]).columns.tolist()
print("\nEstadísticas numéricas:")
display(df[num_cols].describe().T)

# 3) Categóricas: cardinalidad y top valores
cat_cols = df.select_dtypes(include=['object']).columns.tolist()
print("\nColumnas categóricas:", cat_cols)
for c in cat_cols:
    print(f"\n{c} (n_únicos={df[c].nunique()}):")
    print(df[c].value_counts(dropna=False).head(10))

# 4) Sanidad de rangos (negativos o fuera de rango típico)
checks = {
    'CreditScore_negativos': (df['CreditScore'] < 0).sum(),
    'Age_negativos': (df['Age'] < 0).sum(),
    'Tenure_negativos': (df['Tenure'] < 0).sum(),
    'Balance_negativos': (df['Balance'] < 0).sum(),
    'NumOfProducts_negativos': (df['NumOfProducts'] < 0).sum(),
    'EstimatedSalary_negativos': (df['EstimatedSalary'] < 0).sum(),
}
print("\nCheques de valores negativos (no deberían existir):")
print(checks)

# 5) Outliers rápidos por percentiles
q = df[num_cols].quantile([0.01, 0.99])
print("\nPercentiles 1% y 99% (para ojear outliers):")
display(q)






Balance de clases (Exited):
False    7963
True     2037
Name: Exited, dtype: int64

Balance relativo (Exited):
False    0.796
True     0.204
Name: Exited, dtype: float64

Estadísticas numéricas:


count           mean           std     min       25%  \
CreditScore      10000.0     650.528800     96.653299  350.00    584.00   
Age              10000.0      38.921800     10.487806   18.00     32.00   
Tenure           10000.0       4.634300      2.989725    0.00      2.00   
Balance          10000.0   76485.889288  62397.405202    0.00      0.00   
NumOfProducts    10000.0       1.530200      0.581654    1.00      1.00   
EstimatedSalary  10000.0  100090.239881  57510.492818   11.58  51002.11   

                        50%          75%        max  
CreditScore         652.000     718.0000     850.00  
Age                  37.000      44.0000      92.00  
Tenure                4.000       7.0000      10.00  
Balance           97198.540  127644.2400  250898.09  
NumOfProducts         1.000       2.0000       4.00  
EstimatedSalary  100193.915  149388.2475  199992.48


Columnas categóricas: []

Cheques de valores negativos (no deberían existir):
{'CreditScore_negativos': 0, 'Age_negativos': 0, 'Tenure_negativos': 0, 'Balance_negativos': 0, 'NumOfProducts_negativos': 0, 'EstimatedSalary_negativos': 0}

Percentiles 1% y 99% (para ojear outliers):


CreditScore   Age  Tenure      Balance  NumOfProducts  EstimatedSalary
0.01        432.0  21.0     0.0       0.0000            1.0        1842.8253
0.99        850.0  72.0    10.0  185967.9854            3.0      198069.7345

<div class="alert alert-block alert-success">
<b>Comentario del revisor (1ra Iteracion)</b> <a class=“tocSkip”></a>

Correcto! Siempre en proyectos de clasificación utilizando Machine Learning hay que revisar el balance de las clases ya que de esto dependerá las técnicas a aplicar o los modelos a utilizar para conseguir un buen desempeño en las predicciones
</div>

Durante la exploración de datos se identificó que el conjunto está conformado por 10,000 registros y 14 columnas. Se observó un claro desequilibrio de clases en la variable objetivo Exited, con aproximadamente un 80% de clientes que permanecen y un 20% que se van. Variables como RowNumber, CustomerId y Surname no aportan valor predictivo, mientras que Geography y Gender son categóricas relevantes que requieren codificación. Se detectaron valores nulos en Tenure, los cuales deberán imputarse. El resto de variables numéricas (CreditScore, Age, Balance, NumOfProducts, EstimatedSalary) presentan rangos consistentes, sin valores negativos, aunque con outliers naturales propios de los datos financieros. En general, el dataset se encuentra en buen estado para continuar con la preparación y el modelado.



## 2.2 Entrenamiento del modelo excluyendo equilibrio

In [7]:
# 2) OHE rápido para categóricas
df_ohe = pd.get_dummies(df, drop_first=True)

# 3) separar target y features
y = df_ohe['Exited']
X = df_ohe.drop('Exited', axis=1)

# 4) split (estratificado para mantener proporciones)
X_train, X_valid, y_train, y_valid = train_test_split(
    X, y, test_size=0.2, random_state=12345, stratify=y
)

# 5) modelo baseline SIN tener en cuenta el desequilibrio
model = LogisticRegression(max_iter=1000, random_state=12345)  # sin class_weight
model.fit(X_train, y_train)

# 6) predicciones y métricas
proba_valid = model.predict_proba(X_valid)[:, 1]
pred_valid = (proba_valid >= 0.5).astype(int)

print("Baseline (sin tratar desbalance)")
print("F1 :", round(f1_score(y_valid, pred_valid), 3))
print("AUC:", round(roc_auc_score(y_valid, proba_valid), 3))


Baseline (sin tratar desbalance)
F1 : 0.069
AUC: 0.681


El bajo F1 muestra que el modelo no es útil en la práctica si usamos un umbral fijo de 0.5.

El AUC-ROC más alto indica que hay información útil en los datos, solo que el modelo necesita un mejor manejo del desbalance (upsampling, downsampling, class_weight, etc.) y quizá un ajuste de umbral para realmente identificar la clase positiva.

En conclusión: el baseline confirma que el desbalance afecta mucho al F1, y el siguiente paso es aplicar técnicas para corregirlo.

<div class="alert alert-block alert-success">
<b>Comentario del revisor (1ra Iteracion)</b> <a class=“tocSkip”></a>

Correccto, se puede ver que el desbalance afecta el desempeño de los modelos
</div>

# Punto 3

### Upsampling

In [8]:

df_ohe = pd.get_dummies(df, drop_first=True)

y = df_ohe['Exited']
X = df_ohe.drop('Exited', axis=1)

X_train, X_valid, y_train, y_valid = train_test_split(
    X, y, test_size=0.2, random_state=12345, stratify=y
)

In [12]:
# Función de upsampling (replicar clase 1 hasta igualar a la 0)
def upsample_minority(X, y, random_state=12345):
    data = X.copy()
    data['Exited'] = y.values

    maj = data[data['Exited'] == 0]
    mino = data[data['Exited'] == 1]

    # replicar clase 1 hasta el tamaño de la clase 0
    reps = int(len(maj) / len(mino))
    rem  = len(maj) - reps * len(mino)
    mino_rep = pd.concat([mino]*reps + [mino.sample(rem, random_state=random_state)], axis=0)

    up = pd.concat([maj, mino_rep], axis=0)
    up = shuffle(up, random_state=random_state).reset_index(drop=True)

    X_up = up.drop('Exited', axis=1)
    y_up = up['Exited']
    return X_up, y_up

X_train_up, y_train_up = upsample_minority(X_train, y_train)
print("Tamaños tras upsampling:", X_train_up.shape, y_train_up.value_counts().to_dict())


Tamaños tras upsampling: (12740, 11) {False: 6370, True: 6370}


In [13]:
#Entrenar y evaluar
# Modelo con datos upsampled
lr_up = LogisticRegression(max_iter=1000, random_state=12345)
lr_up.fit(X_train_up, y_train_up)

proba_valid = lr_up.predict_proba(X_valid)[:, 1]
pred_valid  = (proba_valid >= 0.5).astype(int)

print("Logistic Regression con upsampling")
print("F1  :", round(f1_score(y_valid, pred_valid), 3))
print("AUC :", round(roc_auc_score(y_valid, proba_valid), 3))



Logistic Regression con upsampling
F1  : 0.454
AUC : 0.732


El F1 ya subió mucho más frente al baseline (0.069). el upsampling ayudó a detectar más casos de Exited=1.

El AUC (0.732) indica que el modelo sí separa razonablemente a ambas clases a nivel de probabilidades (independiente del umbral).

<div class="alert alert-block alert-success">
<b>Comentario del revisor (1ra Iteracion)</b> <a class=“tocSkip”></a>

Perfecto, técnicas cómo sobre-muestreo o sub-muestreo o la modificación de parámetros en algunos modelos para especificar el desbalance ayuda a los modelos a poder generalizar mejor cada caso y obtener un buen desempeño
</div>

In [14]:
# Afianr umbral para maximizar f1

best_t, best_f1 = 0.5, -1
for t in np.arange(0.1, 0.9, 0.01):
    f1 = f1_score(y_valid, (proba_valid >= t).astype(int))
    if f1 > best_f1:
        best_f1, best_t = f1, t

print("Mejor umbral:", round(best_t,2), " | F1:", round(best_f1,3))


Mejor umbral: 0.55  | F1: 0.472


Mover el umbral de 0.50 a 0.55 te dio un pequeño aumento de F1 (+0.018).

Esto significa que, a 0.55, el balance precisión–recall es ligeramente mejor que a 0.50 (probablemente subió la precisión y bajó un poco el recall).

AUC no cambia con el umbral ya que mide ranking de probabilidades

*Conclusión del enfoque 1*:

El upsampling + ajuste de umbral mejora de forma clara el F1, pero aún no alcanza el objetivo del proyecto (≥ 0.59).

### Class weight Balanceado

In [16]:
df_ohe = pd.get_dummies(df, drop_first=True)

y = df_ohe['Exited']
X = df_ohe.drop('Exited', axis=1)

X_train, X_valid, y_train, y_valid = train_test_split(
    X, y, test_size=0.2, random_state=12345, stratify=y
)

# util para buscar el mejor umbral por F1
def best_threshold(proba, y_true, lo=0.1, hi=0.9, step=0.01):
    best_t, best_f1 = 0.5, -1
    for t in np.arange(lo, hi, step):
        f1 = f1_score(y_true, (proba >= t).astype(int))
        if f1 > best_f1:
            best_f1, best_t = f1, t
    return best_t, best_f1


#  1) Logistic Regression con class_weight='balanced' 
lr_bal = LogisticRegression(max_iter=1000, random_state=12345, class_weight='balanced')
lr_bal.fit(X_train, y_train)

proba_lr = lr_bal.predict_proba(X_valid)[:, 1]
pred_lr_050 = (proba_lr >= 0.50).astype(int)

print("LogisticRegression (class_weight='balanced')")
print("F1  @0.50:", round(f1_score(y_valid, pred_lr_050), 3))
print("AUC     :", round(roc_auc_score(y_valid, proba_lr), 3))

t_lr, f1_lr = best_threshold(proba_lr, y_valid, lo=0.2, hi=0.8, step=0.01)
print("Mejor umbral LR:", round(t_lr, 2), "| F1:", round(f1_lr, 3))
print()


#  2) RandomForest con class_weight='balanced' 
rf_bal = RandomForestClassifier(
    n_estimators=300, max_depth=10, random_state=12345, n_jobs=-1,
    class_weight='balanced'
)
rf_bal.fit(X_train, y_train)

proba_rf = rf_bal.predict_proba(X_valid)[:, 1]
pred_rf_050 = (proba_rf >= 0.50).astype(int)

print("RandomForest (class_weight='balanced')")
print("F1  @0.50:", round(f1_score(y_valid, pred_rf_050), 3))
print("AUC     :", round(roc_auc_score(y_valid, proba_rf), 3))

t_rf, f1_rf = best_threshold(proba_rf, y_valid, lo=0.2, hi=0.8, step=0.01)
print("Mejor umbral RF:", round(t_rf, 2), "| F1:", round(f1_rf, 3))

LogisticRegression (class_weight='balanced')
F1  @0.50: 0.46
AUC     : 0.722
Mejor umbral LR: 0.51 | F1: 0.462

RandomForest (class_weight='balanced')
F1  @0.50: 0.643
AUC     : 0.869
Mejor umbral RF: 0.48 | F1: 0.648


LogisticRegression (balanced): F1≈0.46, AUC≈0.72  mejora vs baseline, pero se queda corto.

RandomForest (balanced): F1≈0.64 @0.48, AUC≈0.87  supera holgadamente la meta (≥0.59). Además, el ajuste de umbral (≈0.48) mejora F1 respecto a 0.50.

Conclusión: el mejor enfoque es RandomForest con class_weight='balanced' y umbral ≈ 0.48.

# Punto 4

In [19]:
df_ohe = pd.get_dummies(df, drop_first=True)

y = df_ohe['Exited']
X = df_ohe.drop('Exited', axis=1)

# train/valid/test = 60/20/20
X_temp, X_test, y_temp, y_test = train_test_split(
    X, y, test_size=0.20, random_state=12345, stratify=y
)
X_train, X_valid, y_train, y_valid = train_test_split(
    X_temp, y_temp, test_size=0.25, random_state=12345, stratify=y_temp
)

# reentrenar en train+valid con el mejor modelo
X_trfinal = pd.concat([X_train, X_valid], axis=0)
y_trfinal = pd.concat([y_train, y_valid], axis=0)

best_rf = RandomForestClassifier(
    n_estimators=300, max_depth=10, random_state=12345,
    n_jobs=-1, class_weight='balanced'
)
best_rf.fit(X_trfinal, y_trfinal)

# evaluar en test con el umbral encontrado en valid (≈ 0.48)
proba_test = best_rf.predict_proba(X_test)[:, 1]
thr = 0.48
pred_test = (proba_test >= thr).astype(int)

print("TEST F1 :", round(f1_score(y_test, pred_test), 3))
print("TEST AUC:", round(roc_auc_score(y_test, proba_test), 3))
print(classification_report(y_test, pred_test, digits=3))

TEST F1 : 0.634
TEST AUC: 0.869
              precision    recall  f1-score   support

       False      0.917     0.877     0.896      1593
        True      0.588     0.688     0.634       407

    accuracy                          0.839      2000
   macro avg      0.752     0.782     0.765      2000
weighted avg      0.850     0.839     0.843      2000



Modelo: RandomForestClassifier (class_weight='balanced', umbral ≈ 0.48) entrenado en train+valid y evaluado en test (n=2000).

F1 (test): 0.634  supera el requisito del proyecto (≥ 0.59).

AUC-ROC (test): 0.869  muy buena capacidad de separación entre clases.

Conclusión:

El F1 cumple el umbral y el AUC alto confirma que el modelo asigna probabilidades bien ordenadas (buena separabilidad).

Frente al baseline inicial (F1≈0.069, AUC≈0.681), la mejora es sustancial gracias a class_weight='balanced' y el ajuste de umbral.

# Conclusión


En la evaluación final sobre el conjunto de prueba, el modelo seleccionado —RandomForestClassifier con class_weight='balanced' y umbral de decisión ≈ 0.48— obtuvo F1 = 0.634 y AUC-ROC = 0.869.
El F1 resume el balance entre precisión (0.588) y recall (0.688) para la clase positiva (churn), por lo que es la métrica operativa principal del proyecto; cumplir F1 ≥ 0.59 indica que el modelo identifica una proporción relevante de clientes que se irán sin disparar en exceso los falsos positivos.

Por su parte, el AUC-ROC mide la capacidad de separación entre clases a través de todos los umbrales; un valor de 0.869 confirma que el modelo asigna probabilidades bien ordenadas y que, con ajustes de umbral, es posible moverse a escenarios más conservadores o más sensibles según la necesidad del negocio.

Frente al baseline (Regresión Logística sin correcciones: F1 = 0.069, AUC = 0.681), la mejora es sustancial. Entre los enfoques probados, el upsampling elevó F1 hasta ~0.47, pero el método más efectivo fue ponderar clases (class_weight='balanced') en un Random Forest, además del ajuste de umbral en validación (≈ 0.48), que maximizó F1 manteniendo una AUC alta.

<div class="alert alert-block alert-success">
<b>Comentario del revisor (1ra Iteracion)</b> <a class=“tocSkip”></a>

Te felicito por el trabajo realizado Luis, se nota que conoces las métricas de evaluación y planteas conclusiones muy acertadas en cuanto a los resultados. Es importante siempre en un problema de ML identificar el balance entre las clases de tu variable objetivo para poder escoger la métrica correcta ya que para datasets desbalanceados la métrica de accuracy suele presentar buenos valores pero en realidad las predicciones no son buenas y es por la forma en cómo se calcula esta métrica que puede llevar a interpretaciones erroneas en cambio el f1-score es la ideal para este tipo de casos ya que maneja mejor este tipo de problemas.
    
Saludos!
</div>